gunicorn/arbiter.py

```diff
@@ -494,6 +496,11 @@ def manage_workers(self):
 
     def spawn_worker(self):
         self.worker_age += 1
+        worker_id = -1
+        for i in range(self.num_workers):
```

Comment:
I am not sure I follow the reasoning there. While I understand the need of not using a random hash we can definitely have predictable hashing function. See something like hash(age + appname + ...) . Ideally the hash function could be set via the configuration with good defaults like `hash(age +  module + app)` or using the pid by default . I think it would work for your cases there and would remove any overhead appart the time took to hash the result. Thoughts?